sklearn/base.py

```diff
@@ -254,8 +266,17 @@ def __setstate__(self, state):
         except AttributeError:
             self.__dict__.update(state)
 
+    def _get_tags(self):
+        tags = _DEFAULT_TAGS.copy()
+        for base_class in inspect.getmro(self.__class__):
```

Comment:
Don't we need to reverse this list to give precedence to tags set earlier in the MRO? The precedence should be tested either way.

(I think the official idiom might be `type(self)` rather than `self.__class__` but not sure)

Reply:
hm after thinking about this again, this looks like we're running into the same MRO issue that I was having earlier. I don't think @rth's solution actually works.
I was hoping we would be able to ignore the left-right MRO order with this approach and never overwrite tags. But he have the full tags defined in the ``BaseEstimator``. One solution to make this work is to remove all the tags in the base estimator, not allow overwriting any tags, and then filling in missing tags with the default.